### PR TITLE
sdk v0.45.8, tendermint v0.34.21, iavl v0.19.1 & store options

### PR DESCRIPTION
**Pull request by faddat**
## Description

Evmos v8.0.x is slated to use sdk v0.45.7 and ibc-go v3.2.0 and both of these enable the iavl fast cache features. 

Currently ethermint did not have these and this was causing:

https://github.com/evmos/evmos/issues/868

This patch bumps ethermint's release/v0.19.x branch to:

* cosmos-sdk v0.45.8
* tendermint v0.34.21
* iavl v0.19.1

Mixing fast-cache enabled libraries and non-fast-cache enabled libraries seems not to go well.

## Suggested way forward

* merge this
* cut an ethermint v0.19.1
* We'll take the draft PR I'm making for evmos now, and get rid of the replace statement that I'm using to test the work. 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -292,7 +292,10 @@ func (c TLSConfig) Validate() error {
 
 // GetConfig returns a fully parsed Config object.
 func GetConfig(v *viper.Viper) Config {
-	cfg := config.GetConfig(v)
+	cfg, err := config.GetConfig(v)
```
Comment (faddat) [inline server/config/config.go:295]: Please review the error check here.  There could be a better way to do this.

File: server/indexer_cmd.go [hunk 2]
```diff
@@ -55,7 +55,10 @@ func NewIndexTxCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			stateStore := sm.NewStore(stateDB)
+
+			stateStore := sm.NewStore(stateDB, sm.StoreOptions{
+				DiscardABCIResponses: false,
```
Comment (faddat) [inline server/indexer_cmd.go:60]: I think that this is correct, but please check this, too.

Comment (yihuang) [reply server/indexer_cmd.go:60]: I guess whatever config value is fine here, since it's read-only.

Comment (faddat) [reply server/indexer_cmd.go:60]: Great

File: server/config/config.go (hunk 1 above)
Comment (facs95) [reply server/config/config.go:295]: Change this to return the error instead

**Review by facs95 — APPROVED** [review]
LGTM

**Review by danburck — APPROVED** [review]
LGTM

@faddat can you add a changelog?

**Comment by danburck** [comment]
`Run Gosec` is failing. This is failing on other PRs too and seems to be unrelated to this PR => merging

**Comment by faddat** [comment]
@danburck I think it is because I am making pr's from an external repo

**Pull request merged**